MouseMovementTool.cs

```diff
@@ -0,0 +1,120 @@
+#if UNITY_EDITOR
+using System;
+using System.Collections;
+using System.Collections.Generic;
+//using UnityEditor.Experimental.EditorVR.Core;
+//using UnityEditor.Experimental.EditorVR.Proxies;
+//using UnityEditor.Experimental.EditorVR.UI;
+//using UnityEditor.Experimental.EditorVR.Utilities;
+using UnityEngine;
+//using UnityEngine.InputNew;
+using UnityEngine.UI;
+
+namespace UnityEditor.Experimental.EditorVR.Tools
+{
+    sealed class MouseMovementTool : MonoBehaviour, ITool, ILocomotor, IRayVisibilitySettings,
+        /*ICustomActionMap,*/ ILinkedObject, IUsesViewerScale,
+        IUsesDeviceType, IGetVRPlayerObjects, IBlockUIInteraction, IRequestFeedback
+    {
+        public List<ILinkedObject> linkedObjects
+        {
+            set
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        public Transform cameraRig
+        {
+            private get {
+                return cameraRig;
+            }
+            set
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        //public ActionMap actionMap
+        //{
+        //    get
+        //    {
+        //        throw new NotImplementedException();
+        //    }
+        //}
+
+        public bool ignoreLocking
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        //public void ProcessInput(ActionMapInput input, ConsumeControlDelegate consumeControl)
+        //{
+        //    throw new NotImplementedException();
+        //}
+
+        public GameObject instructions;
```

Comment:
Remove all public fields, replace with private fields, and public properties(where necessary)